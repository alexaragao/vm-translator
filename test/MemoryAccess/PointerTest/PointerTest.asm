@256
D=A
@SP
M=D
// push constant 3030 | line 7
@3030
D=A
@SP
A=M
M=D
@SP
M=M+1
// pop pointer 0 | line 8
@R13
D=A
@R13
M=D
@SP
M=M-1
A=M
D=M
@R13
A=M
M=D
// push constant 3040 | line 9
@3040
D=A
@SP
A=M
M=D
@SP
M=M+1
// pop pointer 1 | line 10
@R13
D=A
@R13
M=D
@SP
M=M-1
A=M
D=M
@R13
A=M
M=D
// push constant 32 | line 11
@32
D=A
@SP
A=M
M=D
@SP
M=M+1
// pop this 2 | line 12
@THIS
D=M
@0
A=D+A
D=A
@R13
M=D
@SP
M=M-1
A=M
D=M
@R13
A=M
M=D
// push constant 46 | line 13
@46
D=A
@SP
A=M
M=D
@SP
M=M+1
// pop that 6 | line 14
@THAT
D=M
@1
A=D+A
D=A
@R13
M=D
@SP
M=M-1
A=M
D=M
@R13
A=M
M=D
// push pointer 0 | line 15
@R13
D=M
@SP
A=M
M=D
@SP
M=M+1
// push pointer 1 | line 16
@R13
D=M
@SP
A=M
M=D
@SP
M=M+1
// add | line 17
@SP
M=M-1
A=M
D=M
@SP
M=M-1
@SP
A=M
M=M+D
@SP
M=M+1
// push this 2 | line 18
@THIS
D=M
@2
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
// sub | line 19
@SP
M=M-1
A=M
D=M
@SP
M=M-1
@SP
A=M
M=M-D
@SP
M=M+1
// push that 6 | line 20
@THAT
D=M
@3
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
// add | line 21
@SP
M=M-1
A=M
D=M
@SP
M=M-1
@SP
A=M
M=M+D
@SP
M=M+1
