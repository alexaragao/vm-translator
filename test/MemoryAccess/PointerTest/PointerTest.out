|RAM[256]| RAM[3] | RAM[4] |RAM[3032|RAM[3046|
|     32 |      0 |      0 |      0 |      0 |
